08_education/college_metrics_county.qmd

```diff
@@ -0,0 +1,1171 @@
+
+Read in the PUMA to county crosswalk file. This file is created by the program generate_puma_county_crosswalks.rmd in the geographic-crosswalks folder. 
+```{r}
+puma_county_crosswalk <- read_csv(here::here("geographic-crosswalks", "data", "crosswalk_puma_to_county.csv"))
+```
+
+## Merge Crosswalk 
+
+Create a variable in the ACS data indicating whether the survey is pre or post-2022. This will be used to join on the correct crosswalk information.
+```{r}
+acs_clean <- acs_clean %>% 
+  mutate(crosswalk_period = ifelse(year < 2022, "pre-2022", "2022"))
+
+```
+
+Check that the assignment worked. All years prior to 2022 should be designated "pre-2022".
+```{r}
+acs_clean %>% 
+  group_by(crosswalk_period, year) %>% 
+  count()
+```
+
+Join the cleaned ACS data onto the crosswalk.
+```{r}
+acs_crosswalked <- left_join(
+    acs_clean, 
+  puma_county_crosswalk,
+  by=c("crosswalk_period", "statefip","puma"), 
+  relationship = "many-to-many"
+)
+```
+
+Drop any observations with NA for afact (i.e. there is no county of interest overlapping this PUMA).
+```{r}
+acs_crosswalked <- acs_crosswalked %>% 
+  drop_na(afact)
+```
+
+Also filter out cases where AFACT is equal to zero. These cases will not be counted in the metric calculation.
+```{r}
+acs_crosswalked <- acs_crosswalked %>% 
+  filter(afact > 0)
+```
+
+Adjust the person weight to account for PUMA-to-county mapping (those where the PUMA is not entirely inside the county).
+
+```{r}
+
+select(acs_crosswalked, perwt, repwtp1, repwtp80, afact)
+acs_crosswalked <- acs_crosswalked %>%
+  mutate(perwt = perwt * afact) %>% 
+  mutate(across(matches("repwtp[0-9]+"), ~.x * afact))
+select(acs_crosswalked, perwt, repwtp1, repwtp80, afact) 
+
+```
+
+## Prepare comuptational variables
+
+Create a binary college_ready variable indicating whether the individual has a HS degree or not (including GED as HS degree equivalent). IPUMS codes  HS graduate or GED (62) and above are all considered to meet this requirement.
+
+```{r}
+acs_crosswalked <- acs_crosswalked %>% 
+  mutate(college_ready = ifelse(education_detailed >= 62, 1, 0)) 
+```
+
+Check how many records, if any, have NA reported for education_detailed after filtering (this should be zero).
+```{r}
+acs_crosswalked %>%
+  filter(is.na(education_detailed)) %>% 
+  group_by(pernum) %>% 
+  count()
+```
+
+Check the distribution of the education_detailed variable one last time - there should be no NA values included. 
+```{r}
+acs_crosswalked %>% 
+  count(education_detailed) %>% 
+  ggplot(mapping = aes(x = factor(education_detailed), y = n)) +
+  geom_col() +
+  theme_minimal() +
+  ggtitle("Education Detailed, All Records") +
+  scale_y_continuous(expand = expansion(mult = c(0, 0.1))) +
+  labs(x = "Educational Attainment",
+       y = NULL)
+```
+
+Confirm there are no vacant properties included in the data.
+```{r}
+acs_crosswalked %>%
+  filter(is.na(vacancy)) %>% 
+  group_by(pernum) %>% 
+  count()
+```
+
+## Create pre-K metric: All
```

Comment:
Here and throughout college metrics scripts, CTRL-F and replace all "pre-K" with "college readiness"--there are a few instances of this.